tests/rptest/tests/datalake/datalake_verifier.py

```diff
@@ -263,9 +268,13 @@ def _verify_next_message(self, partition, iceberg_offset, iceberg_key):
                 self._expected_compacted_keys.add(iceberg_key)
                 return
             else:
+                if self._tolerate_deleted_messages:
+                    return
```

Comment:
Is this mainly for retention? I would have thought compacting keys away is handled by _expected_compacted_keys

Reply:
This is for the case when consumer starts from offset which is > 0, this may happen in case of failures